nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.standard.util.crypto.CipherUtility;
+import org.apache.nifi.processors.standard.util.crypto.KeyedEncryptor;
+import org.apache.nifi.processors.standard.util.crypto.OpenPGPKeyBasedEncryptor;
+import org.apache.nifi.processors.standard.util.crypto.OpenPGPPasswordBasedEncryptor;
+import org.apache.nifi.processors.standard.util.crypto.PasswordBasedEncryptor;
+import org.apache.nifi.security.util.EncryptionMethod;
+import org.apache.nifi.security.util.KeyDerivationFunction;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.Security;
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides functionality of encrypting attributes with various algorithms.
+ * Note. It'll not modify filename or uuid as they are sensitive and are
+ * internally used by either Algorithm itself or FlowFile repo.
+ */
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"encryption", "decryption", "password", "JCE", "OpenPGP", "PGP", "GPG"})
+@CapabilityDescription("Encrypts or Decrypts a FlowFile attributes using either symmetric encryption with a password " +
+        "and randomly generated salt, or asymmetric encryption using a public and secret key.")
+public class EncryptAttributes extends AbstractProcessor {
+
+    public static final String ENCRYPT_MODE = "Encrypt";
+    public static final String DECRYPT_MODE = "Decrypt";
+
+    public static final String WEAK_CRYPTO_ALLOWED_NAME = "allowed";
+    public static final String WEAK_CRYPTO_NOT_ALLOWED_NAME = "not-allowed";
+
+    public static final PropertyDescriptor ATTRIBUTES_TO_ENCRYPT = new PropertyDescriptor.Builder()
+            .name("Attributes to encrypt")
+            .description("Comma separated list of attributes to encrypt, if empty then it'll encrypt all the " +
+                    "attributes including CoreAttributes EXCEPT filename and uuid. " +
+                    "This list is case sensitive and if attribute is not found " +
+                    "then the value will be ignored. " )
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
```

Comment:
Upon further consideration, I think supporting Expression Language and allowing the user to provide a regex here is more robust than a comma-separated list. We already support such operations in `RouteOnAttribute`. With this feature enabled, it would be equally easy to encrypt all _n_ attributes and to encrypt _n - 1_ attributes when _n_ is a large number, while in this case, "all" is very easy but _n - 1_ scales linearly on definition.